cve_bin_tool/checkers/glibc.py

```diff
@@ -17,14 +19,59 @@ class GlibcChecker(Checker):
     CONTAINS_PATTERNS = [
         r"The following command substitution is needed to make ldd work in SELinux",
         r"environments where the RTLD might not have permission to write to the",
+        r"Valid options for the LD_DEBUG environment variable are:",
+        r"To direct the debugging output into a file instead of standard output",
+        r"Mandatory or optional arguments to long options are also mandatory or optional for any corresponding short options.",
     ]
     FILENAME_PATTERNS = [
         r"ldd",
         r"libc6",
         r"glibc-source",
         r"glibc",
         r"nscd",
-        r"libc.so.",
+        r"libc\.so\.6",
+        r"ld-([012]\.[0-9]+)\.so",
+        r"libc-([012]\.[0-9]+)\.so",
+        r"ld-([012]\.[0-9]+\.[0-9]+)\.so",
+        r"libc-([012]\.[0-9]+\.[0-9]+)\.so",
+    ]
+    VERSION_PATTERNS = [
+        r"GNU C Library \(GNU libc\) stable release version ([012]\.[0-9]+)",
+        r"GNU C Library \(GNU libc\) stable release version ([012]\.[0-9]+\.[0-9]+)",
```

Comment:
You can probably combine these two patterns (and some of the other pairs below).  Take a look at the `?` operator or the `{0, n}`style options.  (We haven't done this with all the other checkers, but we probably should on the theory that it could improve speed/accuracy).